staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -2333,12 +2333,13 @@ type Container struct {
 	// Cannot be updated.
 	// +optional
 	WorkingDir string `json:"workingDir,omitempty" protobuf:"bytes,5,opt,name=workingDir"`
-	// List of ports to expose from the container. Exposing a port here gives
-	// the system additional information about the network connections a
-	// container uses, but is primarily informational. Not specifying a port here
+	// List of ports to expose from the container. Not specifying a port here
 	// DOES NOT prevent that port from being exposed. Any port which is
 	// listening on the default "0.0.0.0" address inside a container will be
 	// accessible from the network.
+	// This array might be out-of-date because it merges on containerPort and not (containerPort, protocol).
+	// This information should not be used to get accurate ports exposed by pod.
+	// See https://github.com/kubernetes/kubernetes/issues/103544 for more information.
```

Comment:
That doesn't seem like the right doc fix. I would say "don't use strategic merge patch with this field" and explain why.

Reply:
I think my current contribution makes it clearer that this information should not be taken as truth. Do you want me to add a part indicating "don't use strategic merge patch with this field" or replace my comment by yours ?
I think mine makes it clearer for those not necessarily familiar with merge strategies in k8s. WDYT ?

Reply:
Other than a strategic merge patch or purely client-side confusion, is there any other way for this field to become “out of date”?

If the only API means to get it wrong are strategic merge patch, we can say so.
:thought_balloon: We could _also_, in the API, send a Warning: response any time someone does a strategic merge patch that updates this field (and `kubectl` then prints the warning).

Reply:
> I think my current contribution makes it clearer that this information should not be taken as truth.

But it _is_ the truth. If you have a Service that is delivering to port "`foo`" on a given pod, and someone uses strategic merge patch and accidentally removes that entry from Ports, then the service will stop delivering traffic to that pod. The problem is not that Ports contains incorrect information; it's that the pod has been redefined, and the new definition isn't what you had wanted.

Reply:
Sorry, I should have been clearer in my wording. Indeed you are right, it is the truth.

I don't think there's any other way of the fields `Ports` to be out-of-date.
I agree we should indicate the problem is using strategic merge patch in this case, but I would prefer if we also make it clear for people that are not familiar with it (as was I when I encountered this bug). Is it okay to link k8s docs in those comments ?
I would say something along the lines of 
```
Use of strategic merge patch with this field can lead to errors due to information seemingly being the same but actually being different. This happens when applying diffing with arrays with same port number since the diffing takes only port into account and not the tuple (port, protocol).
See <link to the issue>
See <link to strategic merge patch like https://kubernetes.io/docs/tasks/manage-kubernetes-objects/update-api-object-kubectl-patch/#notes-on-the-strategic-merge-patch>.
```


Sending a warning sounds like a great idea ! Do you mind giving me some pointers on how to do this ? Should I add this to this PR or should I make another one ?
 
I'm also open to suggestions on wording since I'm not a native english speaker.

Reply:
> Sending a warning sounds like a great idea ! Do you mind giving me some pointers on how to do this ? Should I add this to this PR or should I make another one ?

I opened #110627 about that, but it's likely to be complicated. I don't know exactly how the apiserver warning mechanism works...